File: iCyption2022/CftTest/semangat/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
	string numstring;
	cin >> numstring;
	int n = numstring.length();
	int num[n]; // array of digits
	for (int i = 0; i < n; i++)
		num[i] = numstring[i] - '0';
	int sumdigit = 0;
	sumdigit = accumulate(num,num+n,sumdigit);
	if (num[n-1] != 0 && num[n-1] != 5)
		cout << "Lebih Semangat Lagi";
	else{
		if (sumdigit % 9 == 0)
			cout << "Semangat 45";
		else{
			if (n > 1){
				int sumbutfirst = (sumdigit - num[0])% 9;
				int firstdigitnew = 9 - sumbutfirst;
				cout << "Ayo Semangat 45" << "\n";
				cout << "Digit pertama baru: "<< firstdigitnew;
			}
			else
				cout << "Lebih Semangat Lagi";
		}
	}
	return 0;
}

File: iCyption2022/CftTest/semangat/solution.py
def sumOf(n) :
    sum, first = 0, 0
    while (n > 0) :
        sum += n % 10
        first = n
        n //= 10
    return sum, first

n = int(input())

sum, first = sumOf(n)

if (n % 10 != 0 and n % 10 != 5) :
    print("Lebih Semangat Lagi")
else :
    if (sum % 9 == 0) :
        print("Semangat 45")
    else :
        if (n > 9) :
            sumbutfirst = (sum - first) % 9
            firstNew = 9 - sumbutfirst
            print("Ayo Semangat 45")
            print("Digit pertama baru:", firstNew)
        else :
            print("Lebih Semangat Lagi")